### DISCUSSION TITLE
WebGL build size goes up 2.6mb

### DISCUSSION THREAD

**Discussion by kreso22** [KyryloKuzyk/PrimeTween#82] [FAQ] [answered]
Hello :)

Thanks for this amazing tween library!

I have a problem, I wonder if anyone else have run into this.
Initial WebGL build (blank 3d project) is 2.5MB (compressed) and adding PrimeTween adds another 2.6MB. A lot :P

Does this happen to anyone else?

I can reproduce this 100%:
- bring PrimeTween into a blank project (basic 3D, using Unity 2021.3.37f1). 
- delete all the files from the demo folder except the scripts.
- attach Baggage.cs (from demo scripts) to my camera game object (just so PrimeTween is referenced on the scene in the build).
- make the build and it goes up whopping 2.6MB (compressed!).

Build flags: Code optimization: size; IL2CPP Code Generation: Faster (smaller) builds; Stripping: high

Please share any insights as well :)

P.S. I have the same experience with DOTween (it adds 2.7MB).

**Comment by kreso22** [comment, marked as answer]
Upon further inspecting this, I concluded that the size goes up because of including Unity dependencies (for example UI).
So the size goes up, but not because of PrimeTween.

**Reply by KyryloKuzyk** [reply]
Sorry for late reply.
I was about to say the same. PrimeTween references built-in Unity packages like ugui and physics, so when you add PrimeTween, those packages will also be included to the build. 

But those references are soft, meaning that they are not mandatory and you can manually delete them in the manifest.json file, and the build size should be reduced. Can you please try that?

**Reply by kreso22** [reply]
Yes!
To be perfectly clear: there is nothing wrong with PrimeTween! Everything is working as it should.
PrimeTween is great! :-)
